include/swift/SIL/SILBuilder.h

```diff
@@ -526,8 +575,11 @@ class SILBuilder {
 
   UncheckedRefCastInst *createUncheckedRefCast(SILLocation Loc, SILValue Op,
                                                SILType Ty) {
-    return insert(new (F.getModule()) UncheckedRefCastInst(
+    auto *I = insert(new (F.getModule()) UncheckedRefCastInst(
         getSILDebugLocation(Loc), Op, Ty));
+    I->getTypeDefOperandsBuf()[0].set(
+        getOpenedArchetypeDef(Ty, getModule(), OpenedArchetypes));
+    return I;
```

Comment:
@rjmccall That's a good point. Does it mean that we can never reserve a fixed, known-in-advance number of operands for opened archetypes and should do something similar to how we handle `apply` instructions? I.e. we should collect all opened archetypes and then allocate enough memory?